src/main.cpp

```diff
@@ -10,18 +11,36 @@ _Use_decl_annotations_ int WINAPI wWinMain(HINSTANCE hInstance, HINSTANCE hPrevI
 	UNREFERENCED_PARAMETER(pCmdLine);
 	UNREFERENCED_PARAMETER(nCmdShow);
 
-	// We dont want to try and load DLLs from the current directory (Quite often the downloads dir)
-	::SetDefaultDllDirectories(LOAD_LIBRARY_SEARCH_SYSTEM32);
+	const auto kernel32Handle = ::GetModuleHandle(TEXT("kernel32.dll"));
+
+	if (kernel32Handle == NULL) {
+		return -1;
+	}
+
+	// Not supported on all versions of Windows 7
+	typedef BOOL(WINAPI* SetDefaultDllDirectoriesFunc)(DWORD);
+	const SetDefaultDllDirectoriesFunc set_default_dll_directries = reinterpret_cast<SetDefaultDllDirectoriesFunc>(::GetProcAddress(kernel32Handle, "SetDefaultDllDirectories"));
+
+	if (set_default_dll_directries) {
+		// We dont want to try and load DLLs from the current directory (Quite often the downloads dir)
+		set_default_dll_directries(LOAD_LIBRARY_SEARCH_SYSTEM32);
+	}
+
 	// https://docs.microsoft.com/en-us/windows/win32/api/heapapi/nf-heapapi-heapsetinformation
 	::HeapSetInformation(::GetProcessHeap(), HeapEnableTerminationOnCorruption, NULL, 0);
 
-	const auto systemHelper = std::make_shared<SystemHelper>();
+	SystemHelper systemHelper;
+	Logger logger { systemHelper };
+
+	logger.log(L"Fabric launcher native bootstrap log:");
 
-	auto bs = Bootstrap(systemHelper);
+	auto bs = Bootstrap(systemHelper, logger);
 
 	try {
 		bs.launch();
 	} catch (const std::runtime_error& error) {
+		logger.log(L"a runtime error occured:");
+		logger.log(error.what());
 		std::cout << error.what() << std::endl;
```

Comment:
Forgot to remove the old logging method to output the error info?

Reply:
Going to remove all std::cout usage, as its a pain to get even when running via the command line.